python/pyarrow/tests/test_flight.py

```diff
@@ -173,26 +174,30 @@ def get_token(self):
 @contextlib.contextmanager
 def flight_server(server_base, *args, **kwargs):
     """Spawn a Flight server on a free port, shutting it down when done."""
-    # Find a free port
-    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-    with contextlib.closing(sock) as sock:
-        sock.bind(('', 0))
-        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-        port = sock.getsockname()[1]
-
-    auth_handler = kwargs.get('auth_handler')
+    auth_handler = kwargs.pop('auth_handler', None)
+    location = kwargs.pop('location', None)
+
+    if location is None:
+        # Find a free port
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        with contextlib.closing(sock) as sock:
+            sock.bind(('', 0))
+            sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            port = sock.getsockname()[1]
+        location = flight.Location.for_grpc_tcp("localhost", port)
```

Comment:
Ideally we would have a way to let gRPC bind the port and then return it (there is a race condition otherwise). But this needn't be in this PR.